apps/web/src/routes/ChatRoom/ChatRoom.js

```diff
@@ -165,15 +168,23 @@ export default function ChatRoom (props) {
   const postsFuture = useSelector(state => getPosts(state, fetchPostsFutureParams))
   const hasMorePostsFuture = useSelector(state => getHasMorePosts(state, fetchPostsFutureParams))
 
-  const postsForDisplay = useMemo(() => (postsPast || []).concat(postsFuture || []), [postsPast, postsFuture])
+  const postsForDisplay = useMemo(() => {
+    if (!postsPast && !postsFuture) return []
+    return (postsPast || []).concat(postsFuture || [])
+  }, [postsPast, postsFuture])
 
   const fetchPostsPast = useCallback((offset) => {
     if (loadingPast || hasMorePostsPast === false) return Promise.resolve()
     setLoadingPast(true)
-    return dispatch(fetchPosts({ offset, ...fetchPostsPastParams })).then((action) => {
-      messageListRef.current?.data.prepend(action.payload?.data?.group?.posts?.items?.reverse() || [])
-      setLoadingPast(false)
-    })
+    return dispatch(fetchPosts({ offset, ...fetchPostsPastParams }))
+      .then((action) => {
+        const posts = action.payload?.data?.group?.posts?.items
+        if (posts?.length) {
```

Comment:
why was this needed? I think this might break switching to a new chat room that has 0 posts where we need to reset the list of posts to empty